src/WrappedMToken.sol

```diff
@@ -572,15 +573,16 @@ contract WrappedMToken is IWrappedMToken, Migratable, ERC20Extended {
     ) internal view returns (bool isEarning_, uint128 index_, uint112 principal_, uint240 balance_) {
         uint256 unwrapped_ = BalanceInfo.unwrap(_balances[account_]);
 
-        // The most significant 8 bits is always a flag for whether the account is earning or not.
-        isEarning_ = (unwrapped_ >> 248) != 0;
+        // The most significant 1 bit is always a flag for whether the account is earning or not.
+        // The next 15 bits are empty.
+        isEarning_ = (unwrapped_ >> 255) != 0;
 
-        // If the account is not an earner, the 240 least significant bits are simply the present balance.
+        // For a non-earner, the 240 least significant bits are simply the present balance.
         if (!isEarning_) return (isEarning_, uint128(0), uint112(0), uint240(unwrapped_));
 
-        // If the account is an earner, the next 128 bits are the index of the last interaction and the last 112 bits
-        // are the principal amount from which the present balance can be computed.
-        index_ = uint128((unwrapped_ << 8) >> 120);
+        // For an earner, the next 128 bits are the index of the last interaction and the next (and least significant)
+        // 112 bits are the principal amount, from which the present balance can then be computed.
+        index_ = uint128((unwrapped_ << 1) >> 113); // Clear the 1 flag bit then right shift by (1 + 112) bits.
```

Comment:
Do we need to clear the 1 flag bit? Can't we just right shift by `112`?
```suggestion
        index_ = uint128(unwrapped_ >> 112);
```
Shifting right by `112` will clear the principal and casting by `uint128` will retrieve only the index.

Reply:
Great point!